growthstreet_zoo/zoo.py

```diff
@@ -146,15 +103,23 @@ class BaseAnimal(object):
     QUALITY_TOUGH = 'TO'
     QUALITY_TENDER = 'TE'
 
-    def __init__(self, name, cage=None):
+    COMPETITION_MAP = {'Lion': 1,
+                       'Hyena': 2,
+                       'Wildebeest': 3,
+                       'Gazelle': 4}
+
+    def __init__(self, name=None, cage=None):
         """Zoo animals have names, species, cages and are created 'Alive'."""
-        self.name = name
         self.status = self.STATUS_ALIVE
         self.species = None
         self.animal_type = None
         self.sit_rep = None
         self.competition_score = None
         self.meat_quality = None
+        if not name:
```

Comment:
The generate_name code is repeated in a few places.  See if you can use a 'mixin' to collect all this code in one place.